### ISSUE TITLE
GDumb and data augmentation

### ISSUE THREAD

**Issue by gab709** [ContinualAI/avalanche#689]
Hi! I'm testing some strategies to reach the accuracy values stated in the original papers.
In particular, Gdumb with mem_size=500 reaches 90% of accuracy in the split-mnist benchmark.
For the parameters I use the original implementation of GDumb:  https://github.com/drimpossible/GDumb

🐛 **Describe the bug**
There is a drop (~30%) in accuracy using data augmentation. This problem doesn't  appear using other strategies (with the same settings: neural network, parameters, regularization etc.) .

🐜 **To Reproduce**
```
train_transform = Compose([
        #RandomCrop(28, padding=4),
        ToTensor(),
        Normalize((0.1307,), (0.3081,))
    ])
eval_transform = Compose([
        ToTensor(),
        Normalize((0.1307,), (0.3081,))
    ])

scenario = SplitMNIST(
        n_experiences=5, fixed_class_order=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
        seed=1234, train_transform=train_transform, eval_transform=eval_transform)
```
🐝 **Expected behavior**
Data augmentation  should help to achieve better results despite the strategy.

**Comment by AntonioCarta** [comment]
Thanks for the report.
A couple of questions:
- are you using the same hyperparameters? epochs, batch size, learning rate...
- why do you think data augmentation are so important for the failure?

**Comment by gab709** [comment]
> Thanks for the report.
> A couple of questions:
> 
> * are you using the same hyperparameters? epochs, batch size, learning rate...
> * why do you think data augmentation are so important for the failure?

Yes, I'm using the same hyperparameters.
Without data augmentation Gdumb achieves ~87% in accuracy, adding data augmentation the accuracy is ~60%, with same model and same parameters.

**Comment by AndreaCossu** [comment]
Does the original paper use data augmentation? This is a wild guess, but since GDumb only retrains on the replay memory, depending on the kind of data augmentation function, the patterns in the replay memory may not be well representative of the original dataset (like if you shift too much an image, you could miss part of the object).

**Comment by gab709** [comment]
> Does the original paper use data augmentation? This is a wild guess, but since GDumb only retrains on the replay memory, depending on the kind of data augmentation function, the patterns in the replay memory may not be well representative of the original dataset (like if you shift too much an image, you could miss part of the object).

Yes, I use the same data augmentation.

**Comment by AntonioCarta** [comment]
>  Without data augmentation Gdumb achieves ~87% in accuracy, adding data augmentation the accuracy is ~60%, with same model and same parameters.

This is definitely a bug. Maybe GDumb is ignoring the data augmentation when saving the data.

**Comment by AntonioCarta** [comment]
@gab709 if you want to look into this, I can support you. Just ping me on slack if you need help.

Otherwise, I will need the complete script to reproduce the error.

**Comment by AntonioCarta** [comment]
@gab709 is this still an issue? Have you found the problem?

**Comment by gab709** [comment]
> @gab709 is this still an issue? Have you found the problem?

 I think your suggestion is right but I didn't find the problem. I just had a quick look.

**Comment by AntonioCarta** [comment]
@gab709 maybe this is related to #136 ? Are you resetting the model? Can you try it?

**Comment by gab709** [comment]
> @gab709 maybe this is related to #136 ? Are you resetting the model? Can you try it?

Yes, I tried to reset the model (each experience), but the results did not improve. If this is the problem gdumb without augmentation would have had some problems, instead it is ok.

**Comment by AntonioCarta** [comment]
@gab709 I did some experiments. I don't think there is any bug here. The problem is that you are using an MLP. If you use a CNN everything works as expected.

Here are the results using a LeNet-like CNN:
```
┏━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Exp/Task ┃ Top1_Acc_Exp ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━┩
│ E0T0     │ 0.8178       │
│ E1T0     │ 0.9759       │
│ E2T0     │ 0.7593       │
│ E3T0     │ 0.9310       │
│ E4T0     │ 0.6485       │
└──────────┴──────────────┘
```

**Comment by gab709** [comment]
Now it makes sense. Thank you!
